### PR TITLE
Raj Kumar Maurya: Add taxonomic name resolution to the EcoData Retriever to facilitate data science approaches to ecology

### PR DESCRIPTION
**Pull request by raj-maurya**
Checklist for prospect students for Google Summer of Code 2016:

- [x] I contacted NumFOCUS mentors before writing my proposal
- [x] I showed you my contribution (it can be any form: proof-of-concept project idea, some sample code or just a link to your commits from other project)
- [x] I linked to my sample contribution from the proposal
- [x] I linked to my opened issues in numfocus/gsoc repository from the proposal

### CODE REVIEW

**Comment by raj-maurya** [comment]
@ethanwhite  @henrykironde 

Please, review my proposal and suggest some additions or changes.  I would like to know your views on the proposal and waiting for suggestions.

File: 2016/proposals/maurya-kumar-raj.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+The EcoData Retriever is an automated software that is used for finding, downloading, and cleaning up ecological data files, and then organizes the whole data in a local database of your choice. The program cleans and restructures datasets in normalized form and formats them according to the suitable standards of the database management systems.
+
+This project deals with handling the problem of resolving scientific names that are stored in appropriate manner across various datasets. This software would enable its users to conveniently combine different datasets that will be helpful in answering scientific queries. 
+
+## Technical Details
+
+EcoData Retriever is python based tool for downloading, cleaning up, and restructuring ecological data. The information about the repositories are stored in .script files which contain URLs to the files containing the datasets.
+
+This project would build a system for sending a species name to one or more of the taxonomic name resolution services and determine the best name to appear in the generated databases. The project is composed of the following main components:
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: This project would include a system for sending a species name to one or more of the taxonomic name resolution services and determine the best name to appear in the generated databases.

File: 2016/proposals/maurya-kumar-raj.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+* The user interface and the control flow.
+
+The download operation is performed through Engine class, which uses urllib library to fetch the dataset file, and store it locally. This operation happens inside download_file() function in ./lib/engine.py file.
+
+The general problem that we will be dealing would be to maintain the accuracy of datasets for which the names of species are redefined that raises the difficulty in combining the multiple datasets together.
+
+## Schedule of Deliverables
+
+This time line has been made keeping in mind that it will require work at least 40 hours per week. I am not planning on taking any vacations during the summer and would dedicatedly work on the project and also, I will continue to contribute afterwards.
+
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: Include the details in the timeline; for example 
May 25th -  June 7th handling RDFs for testing the XML communications

File: 2016/proposals/maurya-kumar-raj.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+### July 20th - August 2nd
+
+### August 3rd - August 16th
+
+### August 17th - August 21th 19:00 UTC
+
+## Future works
+
+* I will continue to work on the Retriever Project and finish some feature requests and i will be involved in fixing some bugs in Retriever.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: and i will  -> and I will

File: 2016/proposals/maurya-kumar-raj.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+I have worked on few open source projects like Google's tensorflow that deals with Deep Learning and involves the Machine Learning concepts. I also have contribution in fixing bugs in Mozilla community at bugzilla and also contributed as Firefox Student Ambassador.
+
+I have also worked with a start-up from IIT Kanpur,SIDBI Incubation Center, (India) which involved the development of Dynamic UI graph using plot.ly API for Python and Java Script for testing and plotting the large experimental data sets.
+
+I have been the member of NumFocus community and supporting it as a volunteer.
+
+## Academic Experience
+
+I am currently a 3rd year (B.Tech) Computer Science and Engineering student at University Institute of Engineering and Technology, Kanpur, India. I have done some academic projects on database management systems. I am a FOSS enthusiast and like to contribute, collaborate for the same.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: I am a FOSS
I am a  [full name](FOSS)

File: 2016/proposals/maurya-kumar-raj.md (hunk 1 above)
Comment (raj-maurya) [inline 2016/proposals/maurya-kumar-raj.md]: Thank you, your time @henrykironde 

I will fix it ASAP.

File: 2016/proposals/maurya-kumar-raj.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+This project deals with implementation and automating the process of accessing the data in EcoData Retriever that requires the knowledge Python and database systems like MySql, SqLite, which I am familiar with and have experience as well as done few projects on that technology.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: \[some thing like this\]
This project deals with implementing and automating the process of accessing the data with the taxonomic names resolved.
The projects requires the knowledge of Python and database systems like MySql, SqLite.
I have worked on some/similar/familiar projects that had the same requirements and skill set

File: 2016/proposals/maurya-kumar-raj.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+I also have interned at some research organizations like Indian Institute of Science Education Research (IISER) Kolkata, UNESCO.
+Apart from these I have also interned at start-up companies which includes Kanopy Techno Solutions, Pariksha.co, Pollinate Energy, Absentia Virtual Reality.
+
+## Why this project?
+
+While going through the list of organizations and their ideas pages, I was actually looking for the projects which involves the extensive use of database systems, manipulating the large datasets accordingly and I found this project more suitable as well as favorable to me as it involves the usage of almost 80% of the skill sets and technologies that i am already familiar with. Also, my academic as well as non academic projects involved the processing of large data sets using sophisticated database systems accordingly.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: and technologies that i am already familiar     i  -> I

remove (accordingly)

File: 2016/proposals/maurya-kumar-raj.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+I want to be involved in this project because it could help scientists and researchers to do their research and will save their precious time. I will be glad to contribute to this project and help others through this open source projects. So I think it's very meaningful and useful project for me and also i will be learning a lot as well as this project would useful in extending my skills and abilities as well.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: \[this need to be removed  \]
So I think it's very meaningful and useful project for me and also i will be learning a lot as well as this project would useful in extending my skills and abilities as well.
\[or \]
Last, the project will be a stepping stone for expanding my experience, skills and abilities as well

File: 2016/proposals/maurya-kumar-raj.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
+# Add taxonomic name resolution to the EcoData Retriever to facilitate data science approaches to ecology
+
+## Abstract
+
+The EcoData Retriever is an automated software that is used for finding, downloading, and cleaning up ecological data files, and then organizes the whole data in a local database of your choice. The program cleans and restructures datasets in normalized form and formats them according to the suitable standards of the database management systems.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: formats this data according to the suitable

File: 2016/proposals/maurya-kumar-raj.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+This project deals with handling the problem of resolving scientific names that are stored in appropriate manner across various datasets. This software would enable its users to conveniently combine different datasets that will be helpful in answering scientific queries.
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: it's users

File: 2016/proposals/maurya-kumar-raj.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+## Abstract
+
+The EcoData Retriever is an automated software that is used for finding, downloading, and cleaning up ecological data files, and then organizes the whole data in a local database of your choice. The program cleans and restructures datasets in normalized form and formats them according to the suitable standards of the database management systems.
+
+This project deals with handling the problem of resolving scientific names that are stored in appropriate manner across various datasets. This software would enable its users to conveniently combine different datasets that will be helpful in answering scientific queries. 
+
+## Technical Details
+
+EcoData Retriever is python based tool for downloading, cleaning up, and restructuring ecological data. The information about the repositories are stored in .script files which contain URLs to the files containing the datasets.
+
```
Comment (henrykironde) [inline 2016/proposals/maurya-kumar-raj.md]: For the Description of the retriever you can refer to this :
http://www.ecodataretriever.org/
https://retriever.readthedocs.org/en/latest/introduction.html

I suggest giving a description of the problem: Why do we want to add the taxonomic name resolution system to the EcoData Retriever? What is wrong with the Taxonomic names in the data sets?

File: 2016/proposals/maurya-kumar-raj.md (hunk 2 above)
Comment (raj-maurya) [inline 2016/proposals/maurya-kumar-raj.md]: @henrykironde 

I have added the details in the proposal and fixed them as per your suggestions.

Thank You!

File: 2016/proposals/maurya-kumar-raj.md (hunk 5 above)
Comment (raj-maurya) [inline 2016/proposals/maurya-kumar-raj.md]: Done!

File: 2016/proposals/maurya-kumar-raj.md (hunk 10 above)
Comment (raj-maurya) [inline 2016/proposals/maurya-kumar-raj.md]: Thanks for the suggestions @henrykironde

**Comment by raj-maurya** [comment]
Dear @henrykironde 

Please, review my updated proposal, timeline and suggest things that need to include in the proposal as well as any corrections that I need to do.

**Comment by henrykironde** [comment]
I think we are ready for the submission.
Just read through and make sure there is nothing that we may have missed :+1:

**Comment by ethanwhite** [comment]
Looks great @raj-maurya. Go ahead and submit to https://summerofcode.withgoogle.com/ as soon as you have the chance.

**Comment by raj-maurya** [comment]
Sure @ethanwhite  and @henrykironde  
Thank you for your reviews and important suggestions.
I will submit the proposal ASAP to  https://summerofcode.withgoogle.com/

**Comment by raj-maurya** [comment]
@henrykironde and @ethanwhite 

I have submitted the final proposal at https://summerofcode.withgoogle.com/ 
Thank you for your time and suggestions.

You can check the submitted PDF at  https://storage.googleapis.com/summerofcode-prod.appspot.com/gsoc/core_project/doc/1458839765_gitprint__-raj-maurya-gsoc-blob-master-2016-proposals-maurya-kumar-raj.pdf?Expires=1458926178&GoogleAccessId=summerofcode-prod%40appspot.gserviceaccount.com&Signature=mJcnJteZghWzFYNnKgVmM6oVFBTbWBxx1kD28MpkrmEaO1P0IgCWWckVTJhsJJPXVaul7CL2MgFg7Al2JyxWaivSzS5aNAL%2FT8vyN0L7PpdP6LqZ9oJTPaYprzDz8yXsONyTQfs4vuTksNC%2F%2F72BMm%2BzU%2FC4e%2F17TBOmdAY307hFaTH0twIyvsAl32844iSPSBknfkuMzNpXJ47Lz0sf40ZSlM0LqjkZ69Xcqsukfe3S6YfqnOenAzn1p9LC1HtbZWzEuvwsw7Z6fuPjFtx7n0WKBnryC6Zz345yzmF1md1evbVSKV5JXcbDi4zAcA56rloD3Tov5tTw%2FaiiOGmVkQ%3D%3D

**Pull request merged**